### PR TITLE
PR for SRVKS-1174: Document how users can utilize full duplex support for http1

### PR DESCRIPTION
**Pull request by kaldesai**
**Affected versions for cherry-picking:** 

- `serverless-docs-1.33`
- `serverless-docs-1.32`

**Tracking JIRA:** https://issues.redhat.com/browse/SRVKS-1174

**Doc preview:** 

- [Full duplex support for HTTP/1](https://76331--ocpdocs-pr.netlify.app/openshift-serverless/latest/knative-serving/http-configuration/http1-full-duplex-support.html)

I have created an **HTTP configuration** section, which now includes: 

- Global HTTPS redirection
- HTTPS redirection per service
- Full duplex support for HTTP/1 (Newly added)

### CODE REVIEW

**Comment by kaldesai** [comment]
/ serverless

**Comment by kaldesai** [comment]
@skonto is now ready for review. Can you please take a look and let me know?

File: knative-serving/http-configuration/http1-full-duplex-support.adoc [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+.Example `KnativeServing` CR that provides full duplex support for HTTP/1
+[source,yaml]
+----
+apiVersion: serving.knative.dev/v1
+kind: Service
+metadata:
+  name: example-service
+  namespace: default
+  annotations:
+    features.knative.dev/http-full-duplex: "Enabled"
```
Comment (maschmid) [inline knative-serving/http-configuration/http1-full-duplex-support.adoc:22]: The annotation should be on the revision template, not Service,

this was fixed upstream in https://github.com/knative/docs/pull/5839

Comment (skonto) [reply knative-serving/http-configuration/http1-full-duplex-support.adoc:22]: Also the docs are here: https://knative.dev/docs/serving/services/http-protocol/#http1-full-duplex-support-per-workload

**Comment by kaldesai** [comment]
@maschmid @skonto Incorporated your suggestions and changed the content. Can you please take a look?

**Comment by skonto** [comment]
I would add a pointer to the related golang issue. Copying from the upstream docs.

> This should be used in scenarios where the [related Golang issue](https://github.com/golang/go/issues/40747) is hit eg. the application server writes back to QP's reverse proxy before the latter has consumed the whole request.

File: _topic_maps/_topic_map.yml [hunk 2]
```diff
@@ -174,6 +170,15 @@ Topics:
...
     File: using-serving-with-ingress-sharding
+- Name: HTTP configuration
+  Dir: http-configuration
+  Topics:
+  - Name: Global HTTPS redirection
+    File: https-redirect-global
+  - Name: HTTPS redirection per service
+    File: https-redirect-per-service
+  - Name: Full duplex support for HTTP/1
+    File: http1-full-duplex-support
```
Comment (adellape) [inline _topic_maps/_topic_map.yml:181]: Does this need to be a full subdirectory? Especially considering each of the 3 assemblies only have 1 module in each (with no TOCs).

Could this instead just be a single new "HTTP configuration" assembly that `include::`s the 3 modules (the 2 old ones and the 1 new one)? I feel like that would still satisfy the Jira ticket's request to "create another section".

**Review by adellape — COMMENTED** [review]
If you end up changing the topic_map per suggestion, please re-run this through the peer-review queue. Also reminder to squash (and remove WIP from PR title) before merge-review. Thanks!

File: knative-serving/http-configuration/http1-full-duplex-support.adoc [hunk 3]
```diff
@@ -0,0 +1,28 @@
+:_content-type: ASSEMBLY
+include::_attributes/common-attributes.adoc[]
+[id="http1-full-duplex-support"]
+= Full duplex support for HTTP/1
+:context: http1-full-duplex-support
+
+You can enable the HTTP/1 full duplex support for a service by configuring the `features.knative.dev/http-full-duplex` annotation.
+
+NOTE: Verify your HTTP clients before enabling, as older clients may not provide support for HTTP/1 full duplex.
```
[the remark this answers is not in the document]
Comment (kaldesai) [reply knative-serving/http-configuration/http1-full-duplex-support.adoc:9]: Done!

File: _topic_maps/_topic_map.yml (hunk 2 above)
Comment (kaldesai) [reply _topic_maps/_topic_map.yml:181]: Yes, it does need to be a full subdirectory as each assemblies are going to expand by the upcoming next release and will contain the multiple modules. That is why I created like that. :)

**Comment by mburke5678** [comment]
@kaldesai I think this PR is OK to merge. When you are ready, remove the do-not-merge label and let us know.

**Comment by kaldesai** [comment]
@mburke5678 I have removed the do-not-merge label, you can go ahead and merge this. Thank you! :)

**Comment by kaldesai** [comment]
@mburke5678 I have removed the do-not-merge label, you can go ahead and merge this. Thank you! :)

**Pull request merged**